src/sensor_ds18b20.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * (C) Copyright 2021 Tillmann Heidsieck
+ * (C) Copyright 2021 Dominik Laton
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#include "sensor_ds18b20.h"
+
+Sensor_State Sensor_DS18B20::sample() {
+    if (state != SENSOR_NOT_INIT && state != SENSOR_INIT)
+        return state;
+    state = SENSOR_DONE_NOUPDATE;
+
+    Serial.printf("  Sampling sensor DS18B20\n");
+    if (!initialized) {
+        Serial.printf("  DS18B20 not initialized");
+        return SENSOR_NOT_INIT;
+    }
+
+    temp = ds->getTempC();
+
+    if (mem < 0) {
+        state = SENSOR_DONE_UPDATE;
+        return state;
+    }
+
+    ESP.rtcUserMemoryRead(mem, (uint32_t*) &rtc_data, sizeof(rtc_data));
+    if (threshold_helper_float(temp, &rtc_data.temp, 0.1))
+        state = SENSOR_DONE_UPDATE;
+    ESP.rtcUserMemoryWrite(mem, (uint32_t*) &rtc_data, sizeof(rtc_data));
+
+    return state;
```

Comment:
Diese Zeilen sind jetzt in jedem Sensor vorhanden, dass muss man refactor-n

Reply:
Ich würde vorschlagen, wir übernehmen den Punkt in einen Issue, damit wir es nicht vergessen und sehen, wo wir die Struktur anpassen sollten.

Reply:
https://github.com/junkdna/esp8266-sensor-firmware/issues/9